src/geofabrics/processor.py

```diff
@@ -360,7 +376,81 @@ def run(self):
             assert len(bathy_contour_dirs) == 1, f"{len(bathy_contour_dirs)} bathymetry_contours's provided. " + \
                 f"Specficially {catchment_dirs}. Support has not yet been added for multiple datasets."
 
-            if verbose:
+            if self.verbose:
+                print(f"Incorporting Bathymetry: {bathy_contour_dirs}")
+
+            # Load in bathymetry
+            self.bathy_contours = geometry.BathymetryContours(
+                bathy_contour_dirs[0], self.catchment_geometry,
+                z_label=self.get_instruction_general('bathymetry_contours_z_label'),
+                exclusion_extent=self.dense_dem.extents)
+
+            # interpolate
+            self.dense_dem.interpolate_offshore(self.bathy_contours)
+
+        # fill combined dem - save results
+        self.dense_dem.dem.to_netcdf(self.get_instruction_path('result_dem'))
+        self.dense_dem.extents.to_file(self.get_instruction_path('dense_dem_extents'))
+
+
+class OffshoreDemGenerator(BaseProcessor):
+    """ OffshoreDemGenerator executes a pipeline for loading in a Dense DEM and extents before interpolating offshore
+    DEM values. The data and pipeline logic is defined in the json_instructions file.
+
+    The `DemGenerator` class contains several important class members:
+     * catchment_geometry - Defines all relevant regions in a catchment required in the generation of a DEM as polygons.
+     * dense_dem - Defines the hydrologically conditioned DEM as a combination of tiles from LiDAR and interpolated from
+       bathymetry.
+     * bathy_contours - This object defines the bathymetry vectors used by the dense_dem to define the DEM
+       offshore.
+
+    See the README.md for usage examples or GeoFabrics/tests/ for examples of usage and an instruction file
+    """
+
+    def __init__(self, json_instructions: json):
+
+        super(OffshoreDemGenerator, self).__init__(json_instructions=json_instructions)
+
+        self.dense_dem = None
+        self.bathy_contours = None
+
+    def run(self):
+        """ This method executes the geofabrics generation pipeline to produce geofabric derivatives.
+
+        Note it currently only considers one LiDAR dataset. See 'get_lidar_file_list' for where to change this. """
+
+        area_threshold = 10.0/100  # 10%
```

Comment:
It would be great to have some information of the physical meaning of this variable (if not done somewhere else)

Reply:
Good suggestion. I've added something!